r-reproducibility.Rmd

```diff
@@ -0,0 +1,1211 @@
+
+Let's try the `styler::style_file()` function out. Inside a file called
+`non-styled-code.R`, it has:
+
+```{r r-repro-non-styled-code, eval=FALSE}
+# Spacing
+x[,1]
+mean (x, na.rm = TRUE)
+function (x) {}
+height<-feet*12+inches
+sqrt(x ^ 2 + y ^ 2)
+df $ z
+x <- 1 : 10
+
+# Indenting
+if (y < 0 && debug)
+message("Y is negative")
+```
+
+Then we run:
+
+```{r r-repro-use-styler-file, eval=FALSE}
+styler::style_file("testing-styler.R")
+```
+
+```
+Styling  1  files:
+ testing-styler.R ℹ 
+────────────────────────────────────────
+Status	Count	Legend 
+✔ 	0	File unchanged.
+ℹ 	1	File changed.
+✖ 	0	Styling threw an error.
+────────────────────────────────────────
+Please review the changes carefully!
+```
+
+Which changes the file to be styled!
+
+```{r r-repro-styled-code, eval=FALSE}
+# Spacing
+x[, 1]
+mean(x, na.rm = TRUE)
+function(x) {}
+height <- feet * 12 + inches
+sqrt(x^2 + y^2)
+df$ z
+x <- 1:10
+
+# Indenting
+if (y < 0 && debug) {
+  message("Y is negative")
+}
+```
+
+This is more or less everything that the styler package does.
+
+### Exercise: Use styler to fix code formatting
+
+Use the styler package function on the code from the previous exercise by either
+running `styler::style_file()` or with the `"Style selection"` addin when
+highlighting the code.
+
+## Integrating text, code, and results
+
+The most obvious demonstration of reproducibility is when the results obtained
+from executing the analysis code (by an independent entity) are indentical to
+the results presented in the scientific output such as in an article. When there
+is agreement between these two, reproducibility has been more or less acheived.
+In R, there are tools available to *completely* ensure that this happens by
+directly inserting the results from the code *into the scientific ouput*. This is
+done by using [R Markdown][r-markdown], which interweaves R code with text.
+So instead of, for example, manually inserting a figure, you write R code within
+the document to insert the figure for you! Using [R Markdown][r-markdown] can
+save so much time and get your work that much closer to being reproducible.
+
+There are many other advantages to using R Markdown. From the single R Markdown
+format you can use it to create manuscripts, posters, slides, websites, books,
+and many more from simply using R Markdown! In fact, this book was written using
+R Markdown! As a bonus, switching between citation formats or Word templates for
+different journals is easier than doing it with Word.
+
+### Markdown
+
+R Markdown uses, well, [Markdown][markdown] as the format to convert to multiple
+document types. Fun fact: This website is built based on Markdown! While there
+are many "flavours" of Markdown that have been developed over the years, R
+Markdown uses the [Pandoc][pandoc-markdown] version. Pandoc is a combination of
+pan which is Latin for "all" and doc which means document.
+
+Markdown is a "markup language" (like HTML, but simpler) meaning that special
```

Comment:
This is one of those awkward points where we want to introduce a term, but people who don't already know what it means won't understand the definition (they probably haven't seen HTML, so they lack the comparison point).

Reply:
Point taken. Removed.